numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1455,7 +1491,9 @@ _new_argsortlike(PyArrayObject *op, int axis, PyArray_ArgSortFunc *argsort,
         }
     }
 
-    NPY_BEGIN_THREADS_DESCR(descr);
+    if (!needs_api) {
+        NPY_BEGIN_THREADS_DESCR(descr);
```

Comment:
`NPY_BEGIN_THREADS_DESCR` checks the descriptor for the flag, but that is exactly what you do above.  So no need for the `DESCR` part.
(I think the descr isn't changed yet, but if I missed it let me know.)